xapian-core/api/valuerangeproc.cc

```diff
@@ -466,6 +466,17 @@ get_size_multiplier_from_suffix(const char ch)
     }
 }
 
+static bool
+is_valid_size_character(const char ch)
+{
+    // Return true if ch is valid size character ('B','K','M','G')
+    if (ch =='B' || ch =='K' || ch =='M' || ch =='G') {
```

Comment:
This is not what I meant, because now you've just moved the check somewhere else. `get_size_multiplier_from_suffix()` will tell you if it's a valid suffix, so just use that and avoid this duplication.